specs/protocol/deputy-pause-module.md

```diff
@@ -0,0 +1,439 @@
+for all cases in which such a pause would be necessary.
+
+### aSCP-003: Superchain-wide pause authorization process is fast
+
+We assume that the process by which a Superchain-wide pause can be authorized is fast acting such
+that it can make the decision to trigger the pause within a short period of time of any situation
+that would require such a pause.
+
+### aDPM-001: Contract is configured correctly
+
+We assume that all inputs to the contract are configured correctly including:
+
+- Address of the Operations Safe.
+- Address of the `DeputyGuardianModule`.
+- Address of the `SuperchainConfig` contract.
+- Address of the Deputy account.
+
+#### Mitigations
+
+- Verify a signature from the Deputy in the contract constructor.
+- Verify the configured values in the deployment script.
+- Generate and test a signature on testnet.
+- Monitoring if the `DeputyGuardianModule` is not installed into the Security Council Safe.
+
+### aDPM-002: Deputy key is not compromised
+
+We assume that the Deputy key is maintained securely and is not compromised by a malicious actor.
+
+#### Mitigations
+
+- Enforce strong access control around the Deputy key.
+- Monitoring for any usage of the Deputy key.
+- Regular rotation of the Deputy key.
+
+### aDPM-003: Deputy key is not deleted
+
+We assume that the Deputy key has not been deleted by a malicious actor.
+
+#### Mitigations
+
+- Maintain duplicate copies of the key in multiple secure locations.
+- Regularly generate valid signatures from the key.
+
+### aDPM-004: Ethereum will not censor transactions for extended periods of time
+
+We assume that Ethereum will not censor transactions for an extended period of time and that any
+transaction submitted to the network can be processed within a reasonable time bound (e.g., 1h).
+
+#### Mitigations
+
+N/A
+
+### aDPM-005: DeputyGuardianModule pause function correctly triggers pause
+
+We assume that the `pause` function exposed by the `DeputyGuardianModule` will correctly trigger
+the Superchain-wide pause through the Guardian account. We assume that this function will not carry
+out any other action.
+
+#### Mitigations
+
+- Existing audit on the `DeputyGuardianModule`.
+
+### aDPM-006: OpenZeppelin ECDSA and EIP712 contracts are free of critical bugs
+
+We assume that both the `ECDSA` library and the `EIP712` contract provided by OpenZeppelin V4 at
+commit [ecd2ca2c][oz-v4.7.3] (v4.7.3) are free of any critical bugs that would cause their
+behaviors to diverge from their specified behaviors.
+
+#### Mitigations
+
+- Existing audits/safety processes for OpenZeppelin V4.
+
+### aDPM-007: Safe contracts are free of critical bugs
+
+We assume that the Safe contract implementations used by the Security Council Safe and Optimism
+Foundation Operations Safe are free of any critical bugs that would cause their behaviors to
+diverge from their specified behaviors.
+
+#### Mitigations
+
+- Existing audits/safety processes for Safe.
+
+## System Invariants
```

Comment:
random place for this comment: we should indicate somewhere that the DPM is not proxied and why, esp. given that we want/wanted to change the DGM to be proxied.